File: output/cursor/python/easy/q6-valid-anagram.py
# https://leetcode.com/problems/valid-anagram/
# Problem: Valid Anagram

#Problem Description:
#Given two strings s and t, return true if t is an anagram of s, and false otherwise.

# Constraints:
# 1 <= s.length, t.length <= 5 * 10^4
# s and t consist of lowercase English letters.

#Code Structure:
class Solution(object):
    def isAnagram(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        """
        # If lengths are different, they can't be anagrams
        if len(s) != len(t):
            return False
        
        # Count characters in both strings
        char_count = {}
        
        # Count characters in string s
        for char in s:
            char_count[char] = char_count.get(char, 0) + 1
        
        # Subtract characters from string t
        for char in t:
            if char not in char_count:
                return False
            char_count[char] -= 1
            if char_count[char] == 0:
                del char_count[char]
        
        # If all characters are matched, char_count should be empty
        return len(char_count) == 0
    

File: output/sonarqube_analysis/codeium_solutions/java/results/q6-valid-anagram.java
/** https://leetcode.com/problems/valid-anagram */
// Problem: Valid Anagram

//Problem Description:
//Given two strings s and t, return true if t is an anagram of s, and false otherwise.

// Constraints:
// 1 <= s.length, t.length <= 5 * 10^4
// s and t consist of lowercase English letters.

//Code Structure:

class Solution {
    public boolean isAnagram(String s, String t) {
        if (s.length() != t.length()) return false;
        int[] counter = new int[26];
        for (int i = 0; i < s.length(); i++) {
            counter[s.charAt(i) - 'a']++;
            counter[t.charAt(i) - 'a']--;
        }
        for (int c : counter) if (c != 0) return false;
        return true;
    }
}
    